### ISSUE TITLE
FileDialog confirmation button text can't be changed when in folder mode.

### ISSUE THREAD

**Issue by FishyBusinessYT** [godotengine/godot#79803]
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Godot v4.1.1.stable - Windows 10.0.22621 - Vulkan (Mobile) - dedicated NVIDIA GeForce RTX 2060 (NVIDIA; 31.0.15.3640) - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 Threads)

### Issue description

When a FileDialog node is in the "Open Folder" file mode, the "OK Button Text" property can't be changed. Any changes dissapear when toggling its visibility on or after saving while the popup is visible.

### Steps to reproduce

1- Make a new project.
2- Add a FileDialog node.
3- Change the "File Mode" property to "Open Folder".
4- Change the "OK Button Text" property to anything but the default value.
5- The text will revert to the default value if:
- You save while the node is visible.
- You make the node visible.

### Minimal reproduction project

N/A

**Comment by AThousandShips** [comment]
Setting the ok button text is probably not supported like this, but should probably be documented

**Comment by FishyBusinessYT** [comment]
Well, I don't really think that's the case. It's the only mode where this happens, and it was possible to change the ok button text previously.

**Comment by AThousandShips** [comment]
I'm sorry but from a reading of the code you are wrong

The text is set based on the mode, and updated at various times, please look at the source code if you doubt it

**Comment by FishyBusinessYT** [comment]
Yes, but, as stated previously, the ok button text can be changed and saved in other modes. This is an inconsistency at best. Plus, why wouldn't you be able to customize something as simple as a button's text? _If_ it's intended, it would be kind of limiting. 

Also, I thought that the starting button text was updated when switching modes because of, again, consistency between the default values and the file mode.

**Comment by AThousandShips** [comment]
It can be configured theoretically like with the title, but the default behaviour is to communicate the way it works, this should be documented at least as a first step

It's intended, please look at the code and you will see

**Comment by FishyBusinessYT** [comment]
Where can I find said piece of code? 
I'm not one to usually delve into stuff like this.

Also, I agree with you in that the intended behavior should be documented, but I don't think you understand that there's only one mode out of them all that applies to what I'm saying. All the other modes change the ok button text when set, but they don't insist on it if it's changed after.

I'll do some more extensive testing when I get home, so you know what I mean.

**Comment by AThousandShips** [comment]
It will happen when ever you launch the project, because it's designed for that, the relevant code is here:
https://github.com/godotengine/godot/blob/6588a4a29af1621086feac0117d5d4d37af957fd/scene/gui/file_dialog.cpp#L825-L861
https://github.com/godotengine/godot/blob/6588a4a29af1621086feac0117d5d4d37af957fd/scene/gui/file_dialog.cpp#L487-L501
https://github.com/godotengine/godot/blob/6588a4a29af1621086feac0117d5d4d37af957fd/scene/gui/file_dialog.cpp#L459-L481

This is because the button communicates the details

There are two options here, either we add a new option like the one with the title, called `mode_overrides_ok_text` which defaults to true (we have to, or compatibility will be broken) and does the same as the `mode_overrides_title` property

Or we document it

The former seems a bit excessive in my opinion

**Comment by FishyBusinessYT** [comment]
Okay, thanks for the explanation. Although it may seem a bit excessive, maybe it should be done. I'm making a program that can open its own project folders, but they are specific folders with specific structures. 

I'd like to be able to communicate to my users that the folder they're opening can't just be any folder, but a specific kind of project folder.

**Comment by AThousandShips** [comment]
Can't you use the title of the popup for that?

**Comment by FishyBusinessYT** [comment]
Well, of course, and I am currently using that alternative. I wanted to change both things when I ran into this issue. 

In any case, you're right. It seems like this is intended after all. But the way it's done should not only be documented, but made consistent across modes to make the intention apparent.

**Comment by AThousandShips** [comment]
Will start with a documentation and property update today, and discuss the option of a new option like discussed above, but since the dialog doesn't allow you to detect changes easily the update of the button is hard, so I think it's not entirely useful to allow custom ok button text

But will discuss it and see about how to go forward, but will start with a simple solution
